domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java

```diff
@@ -256,4 +266,32 @@ private FixedListMapper() {
             return new ArrayList<>();
         }
     }
+
+    class ComplexFieldsMapper {
+        private ComplexFieldsMapper() {
+            // Default constructor
+        }
+
+        static List<CaseField> map(FieldTypeEntity fieldTypeEntity) {
+            FieldTypeEntity baseFieldTypeEntity = fieldTypeEntity.getBaseFieldType();
+            if (baseFieldTypeEntity == null && isComplexField(fieldTypeEntity.getReference())) {
+                return getCaseFields(fieldTypeEntity.getComplexFields());
+            } else if (baseFieldTypeEntity != null && isComplexField(baseFieldTypeEntity.getReference())) {
+                return getCaseFields(fieldTypeEntity.getComplexFields());
+            }
+            return new ArrayList<>();
+        }
+
+        private static List<CaseField> getCaseFields(List<ComplexFieldEntity> complexFieldEntityList) {
+            List<CaseField> caseFields = new ArrayList<>();
+            for (ComplexFieldEntity complexFieldEntity : complexFieldEntityList) {
+                caseFields.add(EntityToResponseDTOMapper.INSTANCE.map(complexFieldEntity));
```

Comment:
could we refactor to use Streams.map method?

Reply:
DONE